src/libertem/common/udf.py

```diff
@@ -0,0 +1,36 @@
+from enum import Enum
+from typing_extensions import Protocol, TypedDict
+from typing import Union, Literal
```

Comment:
`typing` -> `typing_extensions` for `Literal`, as it was introduced only in Py3.9 IIRC

Reply:
Thx! :-) I'm testing the async PR right now, after that I'll fix it.